src/protocol/OpenConnectionRequest2.php

```diff
@@ -41,7 +40,7 @@ protected function encodePayload(PacketSerializer $out) : void{
 
 	protected function decodePayload(PacketSerializer $in) : void{
 		$this->readMagic($in);
-		if (Cookie::hasServerSecurity()) {
+		if ($this->cookie !== null) {
```

Comment:
The cookie is always going to be null when you haven't decoded it yet

Reply:
> The cookie is always going to be null when you haven't decoded it yet

what should i do?

Reply:
This depends on the protocol. How do we know if the client is going to send a cookie in this request?

Reply:
> This depends on the protocol. How do we know if the client is going to send a cookie in this request?

We can understand something from the length of the incoming buffer, but I don't know if it's the right thing to do.

Reply:
I think we just need a context-aware parser that can read whether cookie is expected. If we always send cookies for all clients, we should always read it. If we never send cookies, we should never read it. It doesn't seem that we require any guessing.